pkg/server/server.go

```diff
@@ -153,6 +156,25 @@ func (pm *PendingDialManager) Remove(random int64) *ProxyClientConnection {
 	return pd
 }
 
+func (pm *PendingDialManager) RemoveForStream(streamUID string) []*ProxyClientConnection {
+	var ret []*ProxyClientConnection
+	if streamUID == "" {
+		return ret
+	}
+	pm.mu.Lock()
+	defer pm.mu.Unlock()
+	for dialID, frontend := range pm.pendingDial {
+		if frontend.frontend == nil {
+			continue
+		}
+		if frontend.frontend.streamUID == streamUID {
```

Comment:
If streamUID is a UID, when would we get multiple matches?

Reply:
Future proofing for the mux case.

And although we added a client library guard https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/452, it is still worth handling here this since 1. the client may be at an older version and 2. the protocol allows receiving multiple DIAL_REQ (and create multiple dials/connections).